docs/src/developers_guide/contributing_code_formatting.rst

```diff
@@ -57,5 +57,12 @@ will look similar to::
 .. note:: You can also run `black`_ and `flake8`_ manually.  Please see the
           their officially documentation for more information.
 
+Type Hinting
+------------
+Iris is gradually adding
+`type hints <https://docs.python.org/3/library/typing.html>`_ into the
+codebase. Where possible please include/add type hints in code that you are
```

Comment:
```suggestion
codebase. If you feel confident to do so, please include/add type hints in code that you are
```

Reply:
I get what you're saying, but I feel it's something worth pushing for to a certain degree, despite the anticipated pain. It has been explained to me that typing is more than a helpful decoration - widespread typing makes the codebase more robust, and even allows for a smaller test suite.

So I'm cautious about making the language **too** forgiving. Can we have a third opinion @SciTools/iris-devs?

Reply:
Can we put it onto the reviewer ?
suggest that reviewer "will consider"

Reply:
Also @bjlittle suggests : especially new code

Reply:
1212d263be1002f7b66e816220c8c9ecf7f907be